lib/OpenQA/WebAPI/Plugin/AMQP.pm

```diff
@@ -70,6 +71,179 @@ sub log_event {
+    }
+    else {
+        # unhandled artifact type
+        return undef;
+    }
+
+    # finally, construct the message content
+    my %msg_data = (
+        contact => {
+            name  => 'Fedora openQA',
+            team  => 'Fedora QA',
+            url   => $baseurl,
+            docs  => 'https://fedoraproject.org/wiki/OpenQA',
+            irc   => '#fedora-qa',
+            email => 'qa-devel@lists.fedoraproject.org',
+        },
+        run => {
+            url => "$baseurl/tests/$job_id",
+            log => "$baseurl/tests/$job_id/file/autoinst-log.txt",
+            id  => $job_id,
+        },
+        artifact => {
+            type => $artifact,
+            id   => $artifact_id,
+        },
+        pipeline => {
+            # per https://pagure.io/fedora-ci/messages/issue/61 this
+            # is meant to be unique per test scenario *and* artifact,
+            # so we construct it out of BUILD and the scenario keys.
+            # 'name' is supposed to be a 'human readable name', well,
+            # this is human readable, so we'll just use it twice
+            id   => join('.', "openqa", $job->BUILD, $job->TEST, $job->MACHINE, $job->FLAVOR, $job->ARCH),
+            name => join('.', "openqa", $job->BUILD, $job->TEST, $job->MACHINE, $job->FLAVOR, $job->ARCH),
+        },
+        test => {
+            # openQA tests are pretty much always validation
+            category => 'validation',
+            # test identifier: test name plus scenario keys
+            type      => join(' ', $job->TEST, $job->MACHINE, $job->FLAVOR, $job->ARCH),
+            namespace => $test_namespace,
+        },
+        system => {
+            # it's interesting whether we should record info on the
+            # *worker host itself* or the *SUT* (the VM run on top of
+            # the worker host environment) here...on the whole I think
+            # SUT is more in line with expectations, so let's do that
+            os => "fedora-${sysrelease}",
+            # openqa provisions itself...we *could* I guess set this
+            # to 'createhdds' if we booted a disk image, but ehhhh
+            provider     => 'openqa',
+            architecture => $job->ARCH,
+            variant      => $job->settings_hash->{SUBVARIANT},
+        },
+        generated_at => $generated_at,
+        version      => "0.2.1",
+    );
+
+    # add keys that don't exist in all cases to the message
+    if ($stdevent eq 'complete') {
+        $msg_data{test}{result} = $job->result;
+        $msg_data{test}{result} = 'info' if $job->result eq 'softfailed';
+    }
+    elsif ($stdevent eq 'error') {
+        $msg_data{error} = {};
+        $msg_data{error}{reason} = $job->result;
+    }
+    elsif ($stdevent eq 'queued') {
+        # this is a hint to consumers that the job probably went away
+        # if they don't get a 'complete' or 'error' in 4 hours
+        # FIXME: we should set this as 2 hours on 'running', but we
```

Comment:
`FIXME` in a submitted PR?

Reply:
It's not fixable until https://progress.opensuse.org/issues/31069 is fixed.